### PR TITLE
Add DB migration for oauth2_unvalidated_credentials table

### PR DESCRIPTION
**Pull request by seanh**
Add a database migration script for the oauth2_unvalidated_credentials
table that has already been added to the model code.

This is our first migration script, so also add migration tooling
(alembic dependency, alembic config file, env.py and script.py.mako
files for alembic) and add a section to the README about how to create a
DB migration.

### CODE REVIEW

File: conf/alembic.ini [hunk 1]
```diff
@@ -0,0 +1,37 @@
+[alembic]
+script_location = lti:migrations
+sqlalchemy.url: postgresql://postgres@localhost:5433/postgres
```
Comment (seanh) [inline conf/alembic.ini:3]: This is the default location for the DB in a dev env, it's overwritten by the `DATABASE_URL` env var in `env.py` below.

File: lti/migrations/env.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+# other values from the config, defined by the needs of env.py,
+# can be acquired:
+# my_important_option = config.get_main_option("my_important_option")
+# ... etc.
+
+
+def get_database_url():
+    if 'DATABASE_URL' in os.environ:
+        return database_url(os.environ['DATABASE_URL'])
+    return config.get_main_option("sqlalchemy.url")
```
Comment (seanh) [inline lti/migrations/env.py]: This is copied from h

File: lti/migrations/env.py [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    This configures the context with just a URL
+    and not an Engine, though an Engine is acceptable
+    here as well.  By skipping the Engine creation
+    we don't even need a DBAPI to be available.
+
+    Calls to context.execute() here emit the given string to the
+    script output.
+
+    """
+    url = get_database_url()
```
Comment (seanh) [inline lti/migrations/env.py:45]: `get_database_url()` is used here, same as in h

File: lti/migrations/env.py [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
+def run_migrations_online():
+    """Run migrations in 'online' mode.
+
+    In this scenario we need to create an Engine
+    and associate a connection with the context.
+
+    """
+    section = config.config_ini_section
+    config.set_section_option(section, 'sqlalchemy.url', get_database_url())
```
Comment (seanh) [inline lti/migrations/env.py:61]: Again stuff copied from h to get it to use `get_database_url()` / the `DATABASE_URL` env var. Apart from these commented changes, this `env.py` file is exactly as generated by `alembic init`.

File: conf/alembic.ini (hunk 1 above)
Comment (seanh) [reply conf/alembic.ini:3]: Apart from editing the `sqlalchemy.url` setting by the way, and deleting some comments, this `alembic.ini` file is as generated by `alembic init`

File: conf/production.ini [hunk 5]
```diff
@@ -29,6 +29,11 @@ level = DEBUG
 handlers =
 qualname = lti
 
+[logger_alembic]
+level = INFO
+handlers =
+qualname = alembic
```
Comment (seanh) [inline conf/production.ini:35]: h had this alembic logging stuff in `app.ini` so I copied it

File: lti/migrations/script.py.mako [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+revision = ${repr(up_revision)}
+down_revision = ${repr(down_revision)}
+
+
+def upgrade():
+    ${upgrades if upgrades else "pass"}
+
+
+def downgrade():
+    ${downgrades if downgrades else "pass"}
```
Comment (seanh) [inline lti/migrations/script.py.mako:23]: This template is the same as `alembic init` generates, same as h's

File: lti/migrations/env.py [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+# other values from the config, defined by the needs of env.py,
+# can be acquired:
+# my_important_option = config.get_main_option("my_important_option")
+# ... etc.
+
+
+def get_database_url():
+    if 'DATABASE_URL' in os.environ:
+        return os.environ['DATABASE_URL']
+    return config.get_main_option("sqlalchemy.url")
```
Comment (seanh) [inline lti/migrations/env.py:30]: This is copied from h except that it just returns `os.environ['DATABASE_URL']` directly, it doesn't have a `database_url()` function to pass it through like h does. Which is maybe not ideal but would have to tweak how the config functions work. This should be fine

**Comment by seanh** [comment]
I've tested that this works, upgrading and downgrading, locally. Also tested that I can break the sqlalchemy URL in alembic.ini and then fix it by setting `DATABASE_URL` in the env, so I think the env var does work for migrations.

File: conf/alembic.ini [hunk 8]
```diff
@@ -0,0 +1,37 @@
+[alembic]
+script_location = lti:migrations
+sqlalchemy.url: postgresql://postgres@localhost:5435/postgres
```
Comment (chdorner) [inline conf/alembic.ini]: I think the port should be `5433`, right?

**Review by chdorner — CHANGES_REQUESTED** [review]
Looks great :+1:

I think the default port in the alembic config is wrong, but otherwise all good.

**Comment by seanh** [comment]
Fixed and merged, thanks!

**Pull request merged**